### PR TITLE
vg-text-input label and icon slots

### PR DESCRIPTION
**Pull request by Austio**
https://www.pivotaltracker.com/story/show/139238377

Adds slots and test for label and icon slots.

### CODE REVIEW

File: src/components/index.js [hunk 1]
```diff
@@ -1,23 +1,9 @@
...
+import buttons from './buttons';
+import TextArea from './text-area';
+import TextInput from './text-input';
 
-const Genome = {
-  Button,
-  ResetButton,
-  SubmitButton,
+export default {
+  ...buttons,
```
Comment (aurelienbottazini) [inline src/components/index.js]: 🏆 this is so much better like that

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import Components from './components';
+
+// Converts pascalCased to pascal-cased
+function genomeName(key) {
```
Comment (aurelienbottazini) [inline src/index.js]: 💡 remove comment and rename to
`convertPascalToKebabCase` ?

File: test/unit/util.js [hunk 3]
```diff
@@ -0,0 +1,119 @@
+/* eslint-disable */
```
Comment (aurelienbottazini) [inline test/unit/util.js]: remove and disable specific lines?

File: .eslintrc.js [hunk 4]
```diff
@@ -31,6 +31,13 @@ module.exports = {
       'vue': 'never'
     }],
     // allow debugger during development
-    'no-debugger': process.env.NODE_ENV === 'production' ? 2 : 0
+    'no-debugger': process.env.NODE_ENV === 'production' ? 2 : 0,
+    // do not error on libraries like sinon that are used in tests
+    'import/no-extraneous-dependencies': ['error', {
+      'devDependencies': [ '**/*.test.js', '**/*.spec.js']
+    }],
+    // without these we will have to do eslint disables on every expect that is expression
```
Comment (Austio) [inline .eslintrc.js]: I agree on these would prefer to have them, the eslint options did not have any configuration that i could find

Good with creating a story to check in future?

**Pull request merged**